# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/20 07:42:05-08:00 torvalds@ppc970.osdl.org 
#   scsi_ioctl: only warn about unknown opcodes if they are truly unknown.
#   
#   We could screw up and warn for opcodes that weren't write-safe and the
#   user tried to use without having write permissions.
#   
#   Not so bad in itself, but that also destroyed the cmd type information.
#   
#   Noticed by Michal Schmidt
# 
# drivers/block/scsi_ioctl.c
#   2005/01/20 07:41:57-08:00 torvalds@ppc970.osdl.org +1 -1
#   scsi_ioctl: only warn about unknown opcodes if they are truly unknown.
#   
#   We could screw up and warn for opcodes that weren't write-safe and the
#   user tried to use without having write permissions.
#   
#   Not so bad in itself, but that also destroyed the cmd type information.
#   
#   Noticed by Michal Schmidt
# 
diff -Nru a/drivers/block/scsi_ioctl.c b/drivers/block/scsi_ioctl.c
--- a/drivers/block/scsi_ioctl.c	2005-01-23 12:05:22 -08:00
+++ b/drivers/block/scsi_ioctl.c	2005-01-23 12:05:22 -08:00
@@ -199,7 +199,7 @@
 			return 0;
 	}
 
-	if (!(type & CMD_WARNED)) {
+	if (!type) {
 		cmd_type[cmd[0]] = CMD_WARNED;
 		printk(KERN_WARNING "scsi: unknown opcode 0x%02x\n", cmd[0]);
 	}
